src/components/tiles/table/use-grid-context.ts

```diff
@@ -49,15 +51,18 @@ export const useGridContext = ({ modelId, showRowLabels, triggerColumnChange }:
   }, [clearSelection, triggerColumnChange]);
 
   const selectRowById = useCallback((rowId: string, select: boolean) => {
-    if (select !== sharedSelection.isSelected(modelId, rowId)) {
-      sharedSelection.select(modelId, rowId, select);
+    if (select !== dataSet.isCaseSelected(rowId)) {
+      clearSelection({ row: false });
+      dataSet.selectCases([rowId], !dataSet.isCaseSelected(rowId));
+      triggerColumnChange();
```

Comment:
```suggestion
      triggerRowChange();
```
This probably isn't entirely clear from the naming, but `triggerRowChange()` is meant to trigger rendering of rows, while `triggerColumnChange()` is meant to trigger a re-evaluation of the column definitions (which should then result in re-rendering the entire table). In the CODAP v3 table this is now handled in a way that makes it easier to only re-render the affected rows rather than having to re-render all rows.